src/actions/search-results.js

```diff
@@ -108,54 +109,106 @@ export function clearFeed () {
   return { type: CLEAR_FEED };
 }
 
+function isTile (item) {
+  return !item.packageOffer;
+}
+
 /**
 * Buffer and show tiles in a mixed fashion
 * to be used by saveSearchResult
 */
 
 export function mixDataInput () {
   // this fn is used to setup a result store, so that we are instance specific.
+  let tiles = [];
+  let packages = [];
+
   let mixture = [];
-  let bufferedResponse = [];
+
   let steps = 1;
-  let highwatermark = 10;
+  let highwatermark = 20;
+
+  let timeout = null;
+
+  function shake () {
+    const total = tiles.length + packages.length;
+    // randomise tiles
+    shuffle(tiles);
+    // sort by rank packages
+
+    // weave tiles and packages to mixture
+    for (let i = 1; i < total + 1; i++) {
+      if (packages.length > 0) {
+        if (i % 6) {
+          tiles.length ? mixture.push(tiles.pop()) : false;
+        }
+        mixture.push(packages.pop());
+      } else {
+        tiles.length ? mixture.push(tiles.pop()) : false;
+      }
+    }
+  }
+
+  function stir (amount) {
+    const pkg = mixture.filter(item => !isTile(item));
+    const ts = mixture.filter(item => isTile(item));
+    packages = pkg;
+    tiles = ts;
+    shake();
+
+    let items = mixture.splice(0, amount);
+    mixture = mixture.filter(item => true);
```

Comment:
Removing undefined and falsy values from the array as splice leaves them in place.